plugins/adapter/onebot/src/bot.ts

```diff
@@ -87,13 +95,32 @@ export class OneBotBot extends Bot<BotConfig> {
   }
 
   async getGuildMember(guildId: string, userId: string) {
+    if (guildId.length > 11) {
+      const memberList = await this.getGuildMemberList(guildId)
+      return memberList.find((member) => member.userId === userId)
+    }
     const data = await this.internal.getGroupMemberInfo(guildId, userId)
     return OneBot.adaptGuildMember(data)
   }
 
   async getGuildMemberList(guildId: string) {
-    const data = await this.internal.getGroupMemberList(guildId)
-    return data.map(OneBot.adaptGuildMember)
+    if (guildId.length > 11) {
+      const { members, bots, admins } = await this.internal.getGuildMembers(guildId)
```

Comment:
建议改成 members = [], ...

Reply:
According to `go-cqhttp` doc, when a specific category is empty, it would become `null` instead of `undefined`. So the initializer would not help here.

Reply:
OK，那么下面的括号是不需要的：

```
...members || [],
```

Reply:
If not, there would be a error like `members is not iteratable`